components/get_link_modal.tsx

```diff
@@ -1,104 +1,101 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
 import SuccessIcon from 'components/widgets/icons/fa_success_icon';
 
-export default class GetLinkModal extends React.PureComponent {
-    static propTypes = {
-        show: PropTypes.bool.isRequired,
-        onHide: PropTypes.func.isRequired,
-        title: PropTypes.string.isRequired,
-        helpText: PropTypes.string,
-        link: PropTypes.string.isRequired,
-    };
+type Props = {
+    show: boolean;
+    onHide: () => void;
+    title: string;
+    helpText?: string;
+    link: string; 
+}
+
+type State = {
+    copiedLink: boolean;
+}
 
-    static defaultProps = {
+export default class GetLinkModal extends React.PureComponent<Props, State> {
+    private textAreaRef = React.createRef<HTMLTextAreaElement>();
+    public static defaultProps = {
         helpText: null,
     };
 
-    constructor(props) {
+    public constructor(props: Props) {
```

Comment:
:Props is not needed here

Reply:
without `:Props` check-types runs into the following error
```
Parameter 'props' implicitly has an 'any' type.

28     public constructor(props) {
```

Reply:
> I also don't understand why it can't infer type of props, since it's clearly stated in type signature of React.PureComponent

I believe this is because the declaration of props in the signature does not define what this class's constructor's parameters are, but rather the parent class `PureComponent` (with the generic type included). Also, it defines what `props` this component takes, but I think that is slightly different than the constructor's parameters on a class/object level, even though we tend to think of them and treat them as one in the same.

As for using `noImplicitAny`, having the flag off makes the compiler too lenient and leads to incorrect or missed typings. Code review needs to be much more strict to catch the missed typings, whereas a strict compiler does much of the work for us. It's meant to discourage you from explicitly using `: any` or `as any` in your code, as that is a sign that your types could be defined/used better.

Reply:
Well put. Let's leave it as is. :+1: